File: android/Controller/GameController.kt
package com.hindtechgroup.sequenceiq.Controller

import com.hindtechgroup.sequenceiq.Model.*

class GameController(private val model: GameModel, private val render: () -> Unit) {
    fun home(){model.screen=Screen.HOME;render()}
    fun difficulty(){model.screen=Screen.DIFFICULTY;render()}
    fun levels(){model.screen=Screen.LEVELS;render()}
    fun selectDifficulty(d:Difficulty){if(model.open(d)){model.difficulty=d;model.screen=Screen.LEVELS;render()}}
    fun start(l:Int){model.start(model.difficulty,l);render()}
    fun answer(v:Int){model.answer(v);render()}
    fun hint(){model.useHint();render()}
    fun retry(){model.retry();render()}
    fun next(){model.next();render()}
    fun navigate(s:Screen){model.screen=s;render()}
}

File: windows/Controller/GameController.cs
using SequenceIQ.Model;

namespace SequenceIQ.Controller;

public sealed class GameController(GameModel model) {
    public GameModel Model => model;
    public void Home() => model.Screen = Screen.Home;
    public void DifficultySelect() => model.Screen = Screen.Difficulty;
    public void Levels() => model.Screen = Screen.Levels;
    public void SelectDifficulty(Difficulty d) { if (model.Open(d)) { model.Difficulty=d; model.Screen=Screen.Levels; } }
    public void Start(int level) => model.Start(model.Difficulty, level);
    public void Answer(int value) => model.Answer(value);
    public void Hint() => model.UseHint();
    public void Retry() => model.Retry();
    public void Next() => model.Next();
    public void Navigate(Screen screen) => model.Screen=screen;
}
